packages/datadog-plugin-aws-sdk/src/services/sqs.js

```diff
@@ -11,6 +11,38 @@ class Sqs {
       'aws.sqs.queue_name': params.QueueName || params.QueueUrl
     })
   }
+
+  responseExtract (params, operation, response, tracer) {
+    if (operation === 'receiveMessage') {
+      if (
+        (!params.MaxNumberOfMessages || params.MaxNumberOfMessages === 1) &&
+        response &&
+        response.Messages &&
+        response.Messages[0]
+      ) {
+        const msg = response.Messages[0]
+        return tracer.extract('text_map', JSON.parse(msg.MessageAttributes._datadog.StringValue))
+      }
+    }
+  }
+
+  requestInject (span, request, tracer) {
+    const operation = request.operation
+    if (operation === 'sendMessage') {
+      if (!request.params) {
+        request.params = {}
+      }
+      if (!request.params.MessageAttributes) {
+        request.params.MessageAttributes = {}
+      }
+      const ddInfo = {}
+      tracer.inject(span, 'text_map', ddInfo)
+      request.params.MessageAttributes._datadog = {
```

Comment:
This is inconsistent with how we usually propagate tracing context and may not be compatible with other languages. Why not simply use `MessageAttributes` as the carrier?

Reply:
`MessageAttributes` is not a simple key-value object. Keys are keys, but values are objects containing type information and also the value itself.

For example, for strings, the value is `{ DataType: 'String', StringValue: 'some string with some stuff in it' }`. This doesn't work with `text_map`. I suppose the `text_map` data could be extracted and transformed into individual properties here. Would that be preferable?